src/sentry/notifications/notifications/base.py

```diff
@@ -147,7 +153,9 @@ def get_referrer(
             referrer += "-" + recipient.__class__.__name__.lower()
         return referrer
 
-    def get_sentry_query_params(self, provider: ExternalProviders, recipient: Team | User) -> str:
+    def get_sentry_query_params(
+        self, provider: ExternalProviders, recipient: Optional[Team | User] = None
+    ) -> str:
```

Comment:
It might be worth leaving a comment about when you wouldn't pass recipient, since it's pretty specific.